app/src/main/java/org/groundplatform/android/ui/common/BindingAdapters.kt

```diff
@@ -52,14 +55,25 @@ object BindingAdapters {
     if (view == null || uri == null) return
 
     try {
-      Picasso.get()
-        .load(uri)
-        .placeholder(R.drawable.ic_photo_grey_600_24dp)
-        .transform(RotateUsingExif(uri, view.context))
-        .into(view)
-    } catch (exception: IllegalStateException) {
-      // Needed for running unit tests
-      Timber.e(exception)
+      val context = view.context
+
+      val bitmap =
```

Comment:
@anandwana001 Is this necessary?

Reply:
yes, for EXIF rotation. we need this.

Reply:
Then we should use https://developer.android.com/reference/android/media/ExifInterface to load the header without loading the whole file.

Reply:
done